sdks/java/extensions/euphoria/euphoria-beam/src/main/java/org/apache/beam/sdk/extensions/euphoria/beam/BeamAccumulatorProvider.java

```diff
@@ -98,63 +125,56 @@ public AccumulatorProvider create(Settings settings) {
   /** Implementation of Counter via {@link org.apache.beam.sdk.metrics.Counter}. */
   public static class BeamMetricsCounter extends BeamMetrics implements Counter {
 
-    public BeamMetricsCounter(String name) {
-      super(name);
+    public BeamMetricsCounter(String namespace, String name) {
+      super(namespace, name);
     }
 
     @Override
     public void increment(long value) {
-      Metrics.counter(this.getClass(), getName()).inc(value);
+      Metrics.counter(getNamespace(), getName()).inc(value);
     }
 
     @Override
     public void increment() {
-      Metrics.counter(this.getClass(), getName()).inc();
+      Metrics.counter(getNamespace(), getName()).inc();
     }
   }
 
   /** Implementation of Histrogram via {@link org.apache.beam.sdk.metrics.Distribution}. */
   public static class BeamMetricsHistogram extends BeamMetrics implements Histogram {
 
-    public BeamMetricsHistogram(String name) {
-      super(name);
+    public BeamMetricsHistogram(String namespace, String name) {
+      super(namespace, name);
     }
 
     @Override
     public void add(long value) {
-      Metrics.distribution(this.getClass(), getName()).update(value);
+      Metrics.distribution(getNamespace(), getName()).update(value);
     }
 
     @Override
     public void add(long value, long times) {
       LongStream.range(0, times)
```

Comment:
I would prefer a simple for loop. It will also save lot of lookups and "concatenations" :) Also we should rethink method signature, I guess `times` param being long is overkill :)
```
public void add(long value, long times) {
  final XXX = Metrics.distribution(getNamespace(), getName());
  for (long i = 0; i < times; i++) {
    XXX.update(i);
  }
}
```

Reply:
fixed